src/rpdk/core/contract/resource_client.py

```diff
@@ -215,28 +215,65 @@ def get_metadata(self):
                 and properties[prop]["insertionOrder"] == "false"
             }
 
-    def transform_model(self, input_model, output_model):
+    def transform_model(self, input_model, output_model):  # pylint: disable=R0914
```

Comment:
It's not related with this PR, But I'm thinking the transform should not be used to transform the model into other model, but should only be used for comparing.  We can do such as `compare(input_model, output_model, transforms)`. The transform semantic here really means comparing, instead of transform A to B.

Other comment, which we have such method in `ResourceClient` we need to have a separated class for model based utils but leave the `ResourceClient` to manage the call with local server.

Reply:
Hey Wenbing, I am not sure I understand the first item. I am happy to set up a chat to talk about them. For now I am merging this CR and can send out a new CR with the suggested changes.